src/submission/submission.service.ts

```diff
@@ -767,9 +774,12 @@ export class SubmissionService {
 
     const { orisCodes } = params;
 
-    return this.submissionQueuePositionRepo
-      .createQueryBuilder('ss')
-      .where('ss.oris_code = ANY(:orisCodes)', { orisCodes })
-      .getMany();
+    return await withSlaveConnection(this.dataSource, async (manager) => {
```

Comment:
Because this is time critical, i.e. it relies on recently updated data, it should not use the read replica.